src/profile-logic/active-tab.js

```diff
@@ -0,0 +1,140 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import { getThreadSelectors } from '../selectors/per-thread';
+import { assertExhaustiveCheck } from '../utils/flow';
+
+import type { State } from '../types/state';
+import type { ThreadIndex, Pid } from '../types/profile';
+import type {
+  GlobalTrack,
+  LocalTrack,
+  TrackIndex,
+} from '../types/profile-derived';
+
+/**
+ * Take the global tracks and decide which one to hide during the active tab view.
+ * Some global tracks are allowed, some tracks are not, and we have to do some
+ * computations for some('process' type specifically).
+ */
+export function computeActiveTabHiddenGlobalTracks(
+  globalTracks: GlobalTrack[],
+  state: State
+): Set<TrackIndex> {
+  const activeTabHiddenGlobalTracks = new Set();
+
+  for (let trackIndex = 0; trackIndex < globalTracks.length; trackIndex++) {
+    const globalTrack: GlobalTrack = globalTracks[trackIndex];
+    const trackType = globalTrack.type;
+
+    switch (trackType) {
+      case 'screenshots':
+        // Do not hide screenshots.
+        break;
+      case 'visual-progress':
+      case 'perceptual-visual-progress':
+      case 'contentful-visual-progress':
+        // Hide those global track types because we want to hide as much as
+        // possible from web developers for now.
+        activeTabHiddenGlobalTracks.add(trackIndex);
+        break;
+      case 'process': {
+        // Do not display empty tracks if the tab filtered thread is empty.
+        if (
+          globalTrack.mainThreadIndex !== undefined &&
+          globalTrack.mainThreadIndex !== null &&
+          isTabFilteredThreadEmpty(globalTrack.mainThreadIndex, state)
+        ) {
+          // Thread is empty and we should hide it.
+          activeTabHiddenGlobalTracks.add(trackIndex);
+        }
+        break;
+      }
+      default:
+        throw assertExhaustiveCheck(trackType, `Unhandled GlobalTrack type.`);
+    }
+  }
+
+  return activeTabHiddenGlobalTracks;
+}
+
+/**
+ * Take the local tracks and decide which one to hide during the active tab view.
+ * Some tracks are not allowed, and we have to do some computations for some('thread' type specifically).
+ */
+export function computeActiveTabHiddenLocalTracksByPid(
```

Comment:
What happens if all the local tracks are hidden, and there are only empty global tracks left?

Edit:
Here is a profile showing what happens: https://perfht.ml/2PzJ4kC

STR:
 * Turn off GeckoMain from the recording
 * Record a profile
 * Check "Show active tab only"

Expected Results:
 * Perhaps show active tab isn't available or checking it will provide an error message.

Actual results:
 * The compositor is still selected, as the selected thread, but is not visible.

This could be a follow-up if you wanted, since it doesn't throw an error, and isn't as common (but does happen in the wild)

Reply:
I know the media team does this, and was also surprised the first time I came across it!

Reply:
Created #2437 to track that.